### PR TITLE
Remove CriticalSection, replace with std::mutex

### PR DESCRIPTION
**Pull request by mvf**
Didn't seem to be used recursively. If it is after all, that's a bug
that needs fixing. If it can't be fixed right away, there's always
std::recursive_mutex to restore the old behavior.

### CODE REVIEW

**Comment by baconpaul** [comment]
Oooh would have been good to chat since there are recursive uses and I’ve never trapped them.

https://github.com/surge-synthesizer/surge/issues/1249 Gives you one example

What happens now if we do recourse? Deadlock?

File: src/common/gui/SurgeGUIEditor.cpp [hunk 1]
```diff
@@ -51,6 +51,10 @@
 #include <stack>
 #include <unordered_map>
 
+#if WINDOWS
+#include <windows.h>
+#endif
```
Comment (baconpaul) [inline src/common/gui/SurgeGUIEditor.cpp:56]: Curious why?

Comment (mvf) [reply src/common/gui/SurgeGUIEditor.cpp:56]: This is for the definitions of `SYSTEMTIME`/`GetSystemTime`. Before, `windows.h` was included implicitly via `SurgeStorage.h`→`CriticalSection.h`.

Comment (baconpaul) [reply src/common/gui/SurgeGUIEditor.cpp:56]: Thanks!
We should move that to std::chrono one day

**Comment by mvf** [comment]
> What happens now if we do recurse? Deadlock?

Yes, which is hopefully easy to debug. Recursive mutexes are a hack and especially bad in audio applications. It seems that the current known recursive use it not intentional though. I'll look into that.

File: src/common/gui/SurgeGUIEditor.cpp (hunk 1 above)
Comment (baconpaul) [reply src/common/gui/SurgeGUIEditor.cpp:56]: we actually didn't need the time at all so just cleaned it up and pushed

**Pull request merged**